### DISCUSSION TITLE
Quic vs TCP

### DISCUSSION THREAD

**Discussion by ryancinsight** [rustdesk/rustdesk#739] [Ideas]
Any thoughts on using Quic instead of TCP? Should be even more secure, I would believe: https://github.com/cloudflare/quiche

**Comment by congzhangzh** [comment]
Quic is udp, right?

maybe UDP is better for multi-media, even on the tunnel of TCP? for my case nomachine is lagging if I disable UDP !

https://github.com/rustdesk/rustdesk/issues/349
https://github.com/rustdesk/rustdesk/issues/713

**Reply by ryancinsight** [reply]
Technically yes, Quic used to stand for quick udp internet connections but seems to be going away from that.

Here's a reference article: https://levelup.gitconnected.com/will-googles-quic-protocol-replace-tcp-6ed991a0ca1e

A couple of items I found intriguing was that everything much be encrypted over it and "QUIC includes a connection ID to the receiver regardless of the source. This allows the connection to be re-established simply by re-sending a single packet, which always contains this ID and that will be considered valid by the receiver even if the sender’s IP address has changed."

Which almost seems like a user account could be setup and used on laptop, desktop, server, or phone that would allow you to send your accounts unique ID and reconnect from any device. 

Overall though it does seem faster and more secure.
